packages/mui-material/src/SpeedDialAction/SpeedDialAction.js

```diff
@@ -173,44 +176,81 @@ const SpeedDialAction = React.forwardRef(function SpeedDialAction(inProps, ref)
 
   const transitionStyle = { transitionDelay: `${delay}ms` };
 
-  const fab = (
-    <SpeedDialActionFab
-      size="small"
-      className={clsx(classes.fab, className)}
-      tabIndex={-1}
-      role="menuitem"
-      ownerState={ownerState}
-      {...FabProps}
-      style={{
+  const externalForwardedProps = {
+    slots,
+    slotProps: {
+      fab: FabProps,
+      ...slotProps,
+    },
+  };
+
+  const [FabSlot, fabSlotProps] = useSlot('fab', {
+    elementType: SpeedDialActionFab,
+    externalForwardedProps,
+    ownerState,
+    shouldForwardComponentProp: true,
+    className: clsx(classes.fab, className),
+    additionalProps: {
+      style: {
         ...transitionStyle,
         ...FabProps.style,
-      }}
-    >
-      {icon}
-    </SpeedDialActionFab>
-  );
+      },
+      tabIndex: -1,
+      role: 'menuitem',
+      size: 'small',
+    },
+  });
+
+  const [TooltipSlot, tooltipSlotProps] = useSlot('tooltip', {
+    elementType: Tooltip,
+    externalForwardedProps,
+    shouldForwardComponentProp: true,
+    ref,
+    additionalProps: {
+      title: tooltipTitle,
+      placement: tooltipPlacement,
+      onClose: handleTooltipClose,
+      onOpen: handleTooltipOpen,
```

Comment:
These should goes into `getSlotProps`, so that `slotProps.tooltip.onOpen` does not replace the `handleTooltipClose`.


```js
    elementType: Tooltip,
    externalForwardedProps,
    shouldForwardComponentProp: true,
    ref,
    getSlotProps: (handlers) => ({
      onClose: () => {
        handlers.onClose?.(),
        handleTooltipClose(),
      },
      onOpen: …
    })
```